src/UnigramWordPredictor.java

```diff
@@ -24,88 +25,145 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
-   * - If the last word in the context is "cat", the next word should be randomly chosen 
-   *   from ["sat", "slept"], giving each an equal 1/2 probability.
+   * - If the last word in the context is "cat", the next word should be randomly
+   * chosen
+   * from ["sat", "slept"], giving each an equal 1/2 probability.
    * 
-   * - If the last word in the context is ".", the next word should be randomly chosen 
-   *   from ["the", "the"], meaning "the" will always be selected 
-   *   since it's the only option.
+   * - If the last word in the context is ".", the next word should be randomly
+   * chosen
+   * from ["the", "the"], meaning "the" will always be selected
+   * since it's the only option.
    * 
-   * - If the last word in the context is "dog", the next word should be "barked" because 
-   *   "barked" is the only word that follows "dog" in the training data.
+   * - If the last word in the context is "dog", the next word should be "barked"
+   * because
+   * "barked" is the only word that follows "dog" in the training data.
    * 
-   * The probabilities of selecting each word should match the relative 
-   * frequencies of the words that follow in the original training data. 
+   * The probabilities of selecting each word should match the relative
+   * frequencies of the words that follow in the original training data.
    * 
    * @param context a list of words representing the current context
    * @return the predicted next word, or null if no prediction can be made
    */
   public String predictNextWord(List<String> context) {
     // TODO: Return a predicted word given the words preceding it
+
+    // As stated in the hint, only the last word in context should be evaluated.
+    String lastWord = context.get(context.size() - 1);
+    // Create empty list for scope purposes
+    List<String> nextPredictedWords = null;
+
+    // check if neighbormap contains lastWord derived from the context
+    if (neighborMap.containsKey(lastWord)) {
+
+      // if lastWord found is true, retrieve list of words that can follow the
+      // last word in the context
+      nextPredictedWords = neighborMap.get(lastWord);
+    }
+    // create instance of random for generating a random output from
+    // nextPredictedWords List
+    Random random = new Random();
     // Hint: only the last word in context should be looked at
-    return null;
+
+    // Used this link to support with the syntaxing of using the random class
+    // https://www.geeksforgeeks.org/generating-random-numbers-in-java/
+
+    // Using the Random class, it will use the size of the nextPredictedWords,
+    // and select a random integer value within the indices of nextPredictedWords.
+    // If the list is ["cat", "cat", "dog"], cat will still have a 2/3 chance
+    // to be selected since cat is given an index twice. In this case, it is 0 =
+    // cat, 1 = cat,
+    // and 2 = dog.
+    return nextPredictedWords.get(random.nextInt(nextPredictedWords.size()));
```

Comment:
This works great! But what would happen if we haven't seen the word in the context before or if it was the last word in the file and duplicated nowhere else? What would your code currently do? What options would you have to deal with this?